python/pyspark/ml/tests.py

```diff
@@ -590,10 +590,14 @@ def test_fit_minimize_metric(self):
         tvsModel = tvs.fit(dataset)
         bestModel = tvsModel.bestModel
         bestModelMetric = evaluator.evaluate(bestModel.transform(dataset))
+        validationMetrics = tvsModel.validationMetrics
 
         self.assertEqual(0.0, bestModel.getOrDefault('inducedError'),
                          "Best model should have zero induced error")
         self.assertEqual(0.0, bestModelMetric, "Best model has RMSE of 0")
+        self.assertEqual(len(grid), len(validationMetrics),
+                         "validationMetrics has the same size of grid parameter")
+        self.assertIn(bestModelMetric, validationMetrics)
```

Comment:
Just a note on this. In general, this is actually not true - the `bestModelMetric` is the metric achieved on the full dataset. Whereas the `max(validationMetrics)` is the metric that the best model achieved on the validation set within `TrainValidationSplit`. It happens to be the same here as we induce a 0 error.

The test is valid though, I thought I would just mention it.

We could also add the same test to `CrossValidator` tests (as again it happens to work for the 0 induced error).